benchmarks/src/Candidates.elm

```diff
@@ -210,3 +210,37 @@ intersperseWithList separator array =
         |> Array.toList
         |> List.intersperse separator
         |> Array.fromList
+
+
+arrayMemberFoldr : a -> Array a -> Bool
+arrayMemberFoldr item array =
+    Array.foldr (\i res -> item == i || res) False array
+
+
+arrayMemberFoldl : a -> Array a -> Bool
+arrayMemberFoldl item array =
+    Array.foldl (\i res -> item == i || res) False array
+
+
+arrayMemberRec : a -> Array a -> Bool
+arrayMemberRec item array =
+    arrayMemberRecHelp 0 item array
+
+
+arrayMemberRecHelp : Int -> a -> Array a -> Bool
+arrayMemberRecHelp n item array =
+    case Array.get n array of
+        Just i ->
+            if i == item then
+                True
+
+            else
+                arrayMemberRecHelp (n + 1) item array
```

Comment:
jsut realized this could be probably "simplified".

```suggestion
            i == item || arrayMemberRecHelp (n + 1) item array
               
```

Reply:
What I'm not sure though is if this could not break tail call optimizer.

Reply:
[I think it does break TCO](https://jfmengels.net/tail-call-optimization/#calls-appearing-in--or--conditions)